backend/notification/socket.js

```diff
@@ -1,13 +1,12 @@
 const { Server } = require("socket.io");
 const http = require("http");
-const { RateLimiterMemory } = require("rate-limiter-flexible");
 const { socket_names } = require("./notif_categories_backend");
+const RateLimiter = require("./rate_limiter");
+
 const FRONTEND_URL = process.env.FRONTEND_URL;
+const rateLimiter = new RateLimiter(60, 5); //5 requests per 60seconds before it gets ratelimited
```

Comment:
I think that having this values as constants is good, but adding a reccomendation. When passing values to a function which are not clear on what is their objective (like in this case the numbers), something that can be done is add the name of the parameter as a comment, like this:

```
const rateLimiter = new RateLimiter(/*time_window=*/60, /*maximum_requests=*/5);
```

Still for this case might be better to store this values as literal constants